framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -863,55 +908,62 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
                                          capture_event->type,
                                          capture_event->type);
 
                     // If we grow too lare on the event vector, it's probably because we have
-                    // received a bunch of events we can not handle.  So remove the first 100
+                    // received a bunch of events we can not handle.  So remove the first 300
                     // events to make room for more without bloating the list size.
                     // TODO: Perhaps do this more elegantly?
-                    if (received_events_.size() > 1000)
+                    if (previously_received_unhandled_events_.size() > 1000)
                     {
-                        GFXRECON_LOG_WARNING("Event list is now %d in size, stripping the first 100!",
-                                             received_events_.size());
-                        received_events_.erase(received_events_.begin(), received_events_.begin() + 100);
+                        GFXRECON_LOG_WARNING(
+                            "Previously received event list is now %d in size, stripping the first 300!",
+                            previously_received_unhandled_events_.size());
+                        previously_received_unhandled_events_.erase(previously_received_unhandled_events_.begin(),
+                                                                    previously_received_unhandled_events_.begin() +
+                                                                        300);
                     }
                 }
                 else if (replay_result == XR_EVENT_UNAVAILABLE)
                 {
-                    // Yield and retry
+                    // No event was ready in the time we waited, so yield and then retry again
                     std::this_thread::sleep_for(std::chrono::nanoseconds(sleep_time));
 
-                    if (sleep_time < kMaxSleepLimitNs)
-                    {
-                        // Next time, sleep for double what we initially slept for.  This way if we're just
-                        // spinning, we spin less and less each time.
-                        sleep_time *= 2;
-                    }
-                    // Clamp it to the max (should stay here from this point forward)
-                    if (sleep_time > kMaxSleepLimitNs)
-                    {
-                        sleep_time = kMaxSleepLimitNs;
-                    }
+                    // Next time, sleep for double what we initially slept for.  This way if we're just
+                    // spinning, we spin less and less each time.
+                    sleep_time *= 2;
                 }
                 else
                 {
+                    // An error of some kind occurred
                     GFXRECON_LOG_ERROR("xrPollEvent encountered an error of type 0x%x", replay_result);
                     break;
```

Comment:
Do we want to break, or just keep retrying?

Reply:
I think for now we should just break on any error.  I think the logic to handle error scenarios other than the expected errors (unavailable, success) result in something else entirely than simple logic we have (because the runtime may have exited) and I don't want to start running down that rabbit trail right now.